infra/base-images/all.sh

```diff
@@ -14,15 +14,55 @@
 # limitations under the License.
 #
 ################################################################################
+#
+# A script to build base images locally.
+#
+# This script is a wrapper around `docker build` that dynamically fetches the
+# official list of images from the Python source of truth, ensuring it never
+# goes out of date.
+#
+# Usage:
+#       # Build the 'latest' version of all images.
+#       ./all.sh
+#
+#       # Build the 'ubuntu-24-04' version of all images.
+#       ./all.sh ubuntu-24-04
+#
+################################################################################
+
+# The first argument is the version tag, e.g., 'latest', 'ubuntu-20-04'.
+VERSION_TAG=${1:-latest}
+
+# Get the directory where this script is located to find the helper script.
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+
+# Fetch the official list of images from the Python source of truth.
+# This avoids duplicating the image list and ensures this script is always
+# up-to-date.
+IMAGE_LIST=$(python3 "${SCRIPT_DIR}/list_images.py")
+
+echo "Building version: ${VERSION_TAG}"
+echo "Images to build: ${IMAGE_LIST}"
+
+# Loop through the official list of images and build each one.
+for image_name in ${IMAGE_LIST}; do
+  image_dir="infra/base-images/${image_name}"
+  
+  if [ "${VERSION_TAG}" == "latest" ]; then
+    dockerfile="${image_dir}/Dockerfile"
+    tag="gcr.io/oss-fuzz-base/${image_name}"
+  else
+    dockerfile="${image_dir}/${VERSION_TAG}.Dockerfile"
+    tag="gcr.io/oss-fuzz-base/${image_name}:${VERSION_TAG}"
+  fi
+
+  if [ ! -f "${dockerfile}" ]; then
+    echo "Skipping build for ${image_name}:${VERSION_TAG} - Dockerfile not found at ${dockerfile}"
+    continue
+  fi
+
+  echo "Building ${tag} from ${dockerfile}..."
+  docker build --pull -t "${tag}" -f "${dockerfile}" "${image_dir}"
```

Comment:
@hunsche Is there a specific reason we added `--pull` here? asking due to https://github.com/google/oss-fuzz/pull/14335

It's causing issues when updating the base images where changes depend on several base images. We should force to use the local version to be consistent with the original functionality IMO